cfn-resources/integration-tests/project/project_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package project
+
+import (
+	"bytes"
+	ctx "context"
+	"log"
+	"os"
+	"path"
+	"testing"
+	"text/template"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	cfn "github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/integration-tests/util"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+type LocalTestContext struct {
+	cfnClient      *cfn.Client
+	atlasClient    *mongodbatlas.Client
+	projectTmplObj TestProject
+	resourceCtx    util.ResourceContext
+	template       string
+	err            error
+}
+
+type TestProject struct {
+	ResourceTypeName string
+	Name             string
+	OrgID            string
+	Profile          string
+	TeamID           string
+	ProjectName      string
+	ProjectID        string
+}
+
+const (
+	resourceTypeName  = "MongoDB::Atlas::Project"
+	resourceDirectory = "project"
+)
+
+var (
+	profile         = os.Getenv("ATLAS_SECRET_PROFILE")
+	orgID           = os.Getenv("ATLAS_ORG_ID")
+	e2eRandSuffix   = util.GetRandNum().String()
+	testProjectName = "cfn-e2e-project" + e2eRandSuffix
+	testTeamName    = "cfn-e2e-team" + e2eRandSuffix
+	stackName       = "stack-project-e2e-" + e2eRandSuffix
+)
+
+func TestProjectCFN(t *testing.T) {
+	teardownSuite, testCtx := setupSuite(t)
+	defer teardownSuite(t)
+
+	t.Run("Validate Template", func(t *testing.T) {
+		testIsTemplateValid(t, testCtx)
+	})
+
+	t.Run("Create Stack", func(t *testing.T) {
+		testCreateStack(t, testCtx)
+	})
+
+	t.Run("Update Stack", func(t *testing.T) {
+		testUpdateStack(t, testCtx)
+	})
+
+	t.Run("Delete Stack", func(t *testing.T) {
+		testDeleteStack(t, testCtx)
+	})
+}
+
+func setupSuite(t *testing.T) (func(t *testing.T), *LocalTestContext) {
+	log.Println("Setting up suite")
```

Comment:
use [t.Log](https://pkg.go.dev/testing#T.Log) and make sure to call https://pkg.go.dev/testing#T.Helper first thing

Reply:
updated